pkg/cephfs/nodeserver.go

```
@@ -102,8 +101,11 @@ func (ns *NodeServer) NodeStageVolume(ctx context.Context, req *csi.NodeStageVol
 		}
 	}
 
-	idLk := nodeVolumeIDLocker.Lock(string(volID))
-	defer nodeVolumeIDLocker.Unlock(idLk, string(volID))
+	if acquired := ns.VolumeLocks.TryAcquire(req.GetVolumeId()); !acquired {
```

(fail fast) move up to line 81